modules/tinymce/src/core/main/ts/api/AddOnManager.ts

```diff
@@ -12,71 +12,73 @@ import Editor from './Editor';
 import I18n from './util/I18n';
 
 /**
- * This class handles the loading of themes/plugins or other add-ons and their language packs.
+ * TinyMCE theme manager class. Subclass of AddOnManager. Allows for a custom theme to be used with TinyMCE.
  *
- * @class tinymce.AddOnManager
- */
-
-/**
- * TinyMCE theme class.
+ * @summary <strong>Warning</strong>: Much of TinyMCE's functionality is provided by the default Silver theme.
+ * Creating a custom theme may require implementing this functionality.
+ * To change TinyMCE's appearance, Tiny recommends changing the Skin instead.
  *
- * @class tinymce.Theme
- */
-
-/**
- * This method is responsible for rendering/generating the overall user interface with toolbars, buttons, iframe containers etc.
+ * @class tinymce.ThemeManager
+ * @example
+ * tinymce.ThemeManager.add('MyTheme', function(editor) {
+ *     // Setup up custom UI elements in the dom
+ *     const div = document.createElement('div');
+ *     const iframe = document.createElement('iframe');
+ *     document.body.appendChild(div);
+ *     document.body.appendChild(iframe);
+ *
+ *     // Themes should fire the SkinLoaded event once the UI has been created
+ *     editor.on('init', () => editor.fire('SkinLoaded'));
  *
- * @method renderUI
- * @param {Object} obj Object parameter containing the targetNode DOM node that will be replaced visually with an editor instance.
- * @return {Object} an object with items like iframeContainer, editorContainer, sizeContainer.
+ *     // Themes must return a renderUI function that returns the editorContainer and iframeContainer
+ *     const renderUI = function() {
+ *         return {
+ *             editorContainer: div,
+ *             iframeContainer: iframe
+ *         };
+ *     };
+ *
+ *     // Return the renderUI function
+ *     return { renderUI };
+ * });
  */
 
 /**
- * Plugin base class, this is a pseudo class that describes how a plugin is to be created for TinyMCE. The methods below are all optional.
+ * TinyMCE plugin manager class. Subclass of AddOnManager. Allows for custom plugins to be added to TinyMCE.
  *
- * @class tinymce.Plugin
+ * @class tinymce.PluginManager
```

Comment:
```suggestion
 * @class tinymce.Plugin
```
Likewise, I don't think we need to document the `PluginManager` directly since it's not a class, but just an instance.